book/src/libs/wasp/components/generator.md

```diff
@@ -0,0 +1,57 @@
+# WASP - Generator
+
+A **Generator** is a component that encapsulates all the characteristics of load generation, including:
+* Load type:
+  * RPS (requests per second)
+  * VUs (virtual users)
+* [Load schedule](./schedule.md)
+* Call logic
+* Response data
+* [Sampling](./sampler.md)
+* Timeouts
+
+> [!WARNING]  
+> RPS load type can only be used with a `Gun`, while VUs can only be used with a `VirtualUser`.
+
+---
+
+### Choosing Between `Gun` and `VirtualUser`
+
+#### **`Gun`**
+* Best for **stateless protocols** (e.g., HTTP).
+* Simplistic in nature; ideal for executing a single operation that does not require setup or teardown.
+* Operates using a **closed model**, where:
+  * The number of requests is fixed.
+  * The load adjusts to meet the target RPS, regardless of the system's response time.
+* Recommended for scenarios focused on measuring **throughput**.
+
+#### **`VirtualUser`**
+* Designed for **stateful protocols** (e.g., `WebSocket`) or workflows involving multiple operations (e.g., authenticating, executing tasks, and logging out).
+* More complex, with dedicated methods for setup and teardown.
+* Operates using an **open model**, where:
```

Comment:
This is a `closed` model. We model the user and if all `VUs` will be stuck in waiting we won't generate more load. The model is `closed`, like `SWIFT`, we do not expect internal DDoS from internal users
Please also change the explanation below, `open` and `closed`

Reply:
To explain it very simply, when we test an `open` system, we do not expect a load feedback loop. We generate the load no matter what because the system is `open` for any user.
In `closed` systems amount of users are usually limited and controlled so we use `VU` and if the system can't handle more all `VUs` will sleep, we won't generate new virtual users
@Tofel

Reply:
yeah, mixed the names, not how they work 🫨